packages/common/utils/src/array/isSubset.ts

```diff
@@ -0,0 +1,20 @@
+/** @tossdocs-ignore */
+
+/**
+ * @name isSubset
+ * @description
+ * Checks if the first array is a subset of the second array
+ *
+ * @param subSet The array to check if it is a subset of the `wholeSet`
+ * @param wholeSet A set of comparison for `subSet`
+ *
+ * @returns boolean ( `subSet` is a subset of `wholeSet` )
+ *
+ * @example
+ * isSubset([], [1,2,3]); // true
+ * isSubset([1,2], [1,2,3]); // true
+ * isSubset([1,2,4], [1,2,3]); // false
+ */
+export function isSubset(subSet: Array<string | number>, wholeSet: Array<string | number>): boolean {
```

Comment:
Will this utility function be used frequently even if we limit the element type of the array to the `string and number` types? I think object type is a lot in real life😄

Can't you implement a utility function that can also determine the subset between objects?

### Example

```tsx
const foo = {
    a:1,
    b:2
}

const foo2 = {
    a:1,
    b:2
}

const foo3 = {
    a:1,
    b:3
}
console.log(isSubset([foo],[foo2,foo3]))
```

Reply:
I agree about this, but as I commented https://github.com/toss/slash/pull/285#discussion_r1253923197

> If purpose of this function is compare of whole unknown type, you should make isEqual methods for unknown type such as lodash.isEqual

If this function want to correspond to more than the `object` type, it is inevitable to create a function such as isEqual.

So, only `string | number` type arguments is not much more convenient, `isEqual` like function must be created first.

Reply:
@ChanhyukPark-Tech
I appreciate your review!

in `@toss/react/useFunnel` use `fast-deep-equal` library for object comparison.

Similarly, it would be nice to use the `fast-deep-equal` library for object comparison for the `isSubset` function.

What do you think?

Reply:
@kim-elijah-sol sorry for late answer

I think `fast-deep-equal` is right solution.

is it good to accept only Array type ? how about accept Iterable type? is it over spec??]

this is my pseudocode

```tsx
import { equal } from 'fast-deep-equal';

function isSubset<T>(iterable1: Iterable<T>, iterable2: Iterable<T>): boolean {
  const array1 = Array.from(iterable1);
  const array2 = Array.from(iterable2);

  return array1.every(item => array2.some(otherItem => equal(item, otherItem)));
}

```

Reply:
I agree that receiving all `Iterable` types is over-spec.
I think it'll be okay to just get the array!